magicschoolbus/Assets/_Code/Tests/TestMovement.cs

```diff
@@ -0,0 +1,33 @@
+﻿using UnityEngine;
+
+namespace MSB
+{
+    [RequireComponent(typeof(CharacterController))]
+    public class TestMovement : MonoBehaviour
+    {
+        public float m_moveSpeed = 3f;
+        public Transform CameraTransform;
+
+        private CharacterController m_CharacterController;
+
+        private void Awake()
+        {
+            m_CharacterController = GetComponent<CharacterController>();
+        }
+
+        private void Update()
```

Comment:
Something weird happens when the character moves directly backwards.  The camera shakes violently, and there are a few seconds where it rotates randomly left or right before setting in when attempting to move straight backwards.  This needs to be looked into, not sure what is causing it, but I'm guessing it is something with remapping the input through the camera's relative position?  The movement looks smooth in the scene view, but the camera object still shakes so I am led to believe the issue is somehow related to that.

Reply:
You're right! The remapping is causing the player to always move directly towards the camera every frame, regardless of how much the camera tries to move out of the way the previous frame. And each frame, due to floating point precision/stability issues, the direction the camera tries to move each frame might be different. I think a small bias can fix this.